src/langchain_google_spanner/vector_store.py

```diff
@@ -0,0 +1,1181 @@
+        content_column: str = CONTENT_COLUMN_NAME,
+        embedding_column: str = EMBEDDING_COLUMN_NAME,
+        metadata_columns: Optional[List[TableColumn]] = None,
+        primary_key: Optional[str] = None,
+        vector_size: Optional[int] = None,
+        secondary_indexes: Optional[List[SecondaryIndex]] = None,
+    ) -> bool:
+        """
+        Initialize the vector store new table in Google Cloud Spanner.
+
+        Parameters:
+        - instance_id (str): The ID of the Spanner instance.
+        - database_id (str): The ID of the Spanner database.
+        - table_name (str): The name of the table to initialize.
+        - client (Client): The Spanner client. Defaults to Client(project="span-cloud-testing").
+        - id_column (str): The name of the row ID column. Defaults to ID_COLUMN_NAME.
+        - content_column (str): The name of the content column. Defaults to CONTENT_COLUMN_NAME.
+        - embedding_column (str): The name of the embedding column. Defaults to EMBEDDING_COLUMN_NAME.
+        - metadata_columns (Optional[List[Tuple]]): List of tuples containing metadata column information. Defaults to None.
+        - vector_size (Optional[int]): The size of the vector. Defaults to None.
+        """
+
+        instance = client.instance(instance_id)
+
+        if not instance.exists():
+            raise Exception("Instance with id-{} doesn't exist.".format(instance_id))
+
+        database = instance.database(database_id)
+
+        if not database.exists():
+            raise Exception("Database with id-{} doesn't exist.".format(database_id))
+
+        database.reload()
+
+        ddl = SpannerVectorStore._generate_sql(
+            database.database_dialect,
+            table_name,
+            id_column,
+            content_column,
+            embedding_column,
+            metadata_columns,
+            primary_key,
+            secondary_indexes
+        )
+
+        operation = database.update_ddl(ddl)
+
+        print("Waiting for operation to complete...")
+        operation.result(100000)
+
+        return True
+
+    @staticmethod
+    def _generate_sql(
+        dialect,
+        table_name,
+        id_column,
+        content_column,
+        embedding_column,
+        column_configs,
+        primary_key,
+        secondary_indexes: Optional[List[SecondaryIndex]] = None,
+    ):
+        """
+        Generate SQL for creating the vector store table.
+
+        Parameters:
+        - dialect: The database dialect.
+        - table_name: The name of the table.
+        - id_column: The name of the row ID column.
+        - content_column: The name of the content column.
+        - embedding_column: The name of the embedding column.
+        - column_names: List of tuples containing metadata column information.
+
+        Returns:
+        - str: The generated SQL.
+        """
+        create_table_statement = f"CREATE TABLE {table_name} (\n"
```

Comment:
Does table name need to be quoted so hyphens can be used

Reply:
it's not required as Spanner doesn't support hyphens.